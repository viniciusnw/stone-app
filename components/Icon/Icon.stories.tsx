import React from "react";
import { Image as XImage } from "expo-image";
import { View, Text, Image as RImage } from "react-native";
import styled from "styled-components/native";
import { Icon, StorybookWrapper } from "@Components";
import { icons } from "@Assets";
import { ScrollView } from "react-native";

const Wrapper = styled(ScrollView)`
  width: 100%;
`;

const IconBox = styled(View)`
  border: 1px solid #ddd;
  padding: 10px;
  margin: 5px;
  width: 110px;
  height: 110px;
  justify-content: center;
  align-items: center;
`;

const IconText = styled(Text)`
  margin: 10px 0 0;
  background-color: #eee;
  border-radius: 3px;
  padding: 2px 6px;
`;

const pack = [
  "left",
  "right",
  "more",
  "menu",
  "list",
  "calendar",
  "plus",
  "wallet",
  "user",
  "security",
  "map",
  "settings",
  "logout",
  "creditcard",
];

export const Default = {
  args: {
    type: "default",
  },
};

export const Image = {
  args: {
    type: "image",
  },
};

export default {
  title: "Icon",
  argTypes: {
    type: {
      options: ["default", "image"],
      control: { type: "radio" },
    },
  },
  component: ({ type }: any) => {
    switch (type) {
      case "default":
        return (
          <StorybookWrapper>
            <Wrapper
              contentContainerStyle={{
                justifyContent: "center",
                flexDirection: "row",
                flexWrap: "wrap",
              }}
            >
              {pack.map((icon) => (
                <IconBox>
                  <Icon size={40} icon={icon} />
                  <IconText>{icon}</IconText>
                </IconBox>
              ))}
            </Wrapper>
          </StorybookWrapper>
        );
      case "image":
        return (
          <StorybookWrapper>
            <Wrapper
              contentContainerStyle={{
                justifyContent: "center",
                flexDirection: "row",
                flexWrap: "wrap",
              }}
            >
              {Object.keys(icons).map((icon) => (
                <IconBox>
                  <RImage
                    style={{ width: 42, height: 42 }}
                    // @ts-ignore
                    source={icons[icon]}
                  />
                  <IconText>{icon}</IconText>
                </IconBox>
              ))}
            </Wrapper>
          </StorybookWrapper>
        );
    }
  },
};
