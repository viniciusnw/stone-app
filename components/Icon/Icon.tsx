import React from "react";
import AntDesign from "react-native-vector-icons/AntDesign";
import Entypo from "react-native-vector-icons/Entypo";
import Feather from "react-native-vector-icons/Feather";
import Ionicons from "react-native-vector-icons/Ionicons";
import MaterialIcons from "react-native-vector-icons/MaterialIcons";
import FontAwesome from "react-native-vector-icons/FontAwesome";

const setIcon = (icon: string, props: any) => {
  switch (icon) {
    case "left":
      return <AntDesign name="leftcircleo" {...props} />;
    case "right":
      return <AntDesign name="rightcircleo" {...props} />;
    case "more":
      return <Entypo name="dots-three-vertical" {...props} />;
    case "menu":
      return <Feather name="menu" {...props} />;
    case "list":
      return <Ionicons name="ios-list" {...props} />;
    case "calendar":
      return <AntDesign name="calendar" {...props} />;
    case "plus":
      return <Entypo name="plus" {...props} />;
    case "wallet":
      return <MaterialIcons name="attach-money" {...props} />;
    case "user":
      return <Feather name="user" {...props} />;
    case "security":
      return <Feather name="lock" {...props} />;
    case "map":
      return <Feather name="map-pin" {...props} />;
    case "settings":
      return <Feather name="settings" {...props} />;
    case "logout":
      return <AntDesign name="logout" {...props} />;
    case "creditcard":
      return <MaterialIcons name="credit-card" {...props} />;
    case "home":
      return <Entypo name="home" {...props} />;
    case "cart":
      return <Entypo name="shopping-cart" {...props} />;
    case "star":
      return <FontAwesome name="star" {...props} />;
    case "staro":
      return <FontAwesome name="star-o" {...props} />;
    case "starh":
      return <FontAwesome name="star-half" {...props} />;
    default:
      return <></>;
  }
};

const Icon = ({ icon, ...props }: any) => {
  return setIcon(icon, props);
};

export default Icon;
