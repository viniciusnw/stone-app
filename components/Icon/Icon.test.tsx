import React from "react";
import renderer from "react-test-renderer";
import { darkTheme, lightTheme } from "@Theme";
import { ThemeProvider } from "styled-components/native";
import "jest-styled-components";
import Icon from "./Icon";

jest.useFakeTimers();

test("Icon", () => {
  const tree = renderer
    .create(
      <ThemeProvider theme={darkTheme}>
        <Icon size={40} icon="more" />
      </ThemeProvider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
