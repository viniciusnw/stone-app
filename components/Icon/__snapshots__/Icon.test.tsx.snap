// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon 1`] = `<Text />`;
